rali/rali/source/meta_node_ssd_random_crop.cpp

```diff
@@ -0,0 +1,211 @@
+    _y2_val.resize(_batch_size);
+}
+void SSDRandomCropMetaNode::update_parameters(MetaDataBatch *input_meta_data)
+{
+    initialize();
+    if (_batch_size != input_meta_data->size())
+    {
+        _batch_size = input_meta_data->size();
+    }
+    set_threshold(_node->get_threshold());
+    _meta_crop_param = _node->get_crop_param();
+    _dst_width = _node->get_dst_width();
+    _dst_height = _node->get_dst_height();
+    _crop_width = _meta_crop_param->cropw_arr;
+    _crop_height = _meta_crop_param->croph_arr;
+    _x1 = _meta_crop_param->x1_arr;
+    _y1 = _meta_crop_param->y1_arr;
+    _x2 = _meta_crop_param->x2_arr;
+    _y2 = _meta_crop_param->y2_arr;
+    vxCopyArrayRange((vx_array)_crop_width, 0, _batch_size, sizeof(uint), _crop_width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_crop_height, 0, _batch_size, sizeof(uint), _crop_height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_x1, 0, _batch_size, sizeof(uint), _x1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y1, 0, _batch_size, sizeof(uint), _y1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_x2, 0, _batch_size, sizeof(uint), _x2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y2, 0, _batch_size, sizeof(uint), _y2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+
+    area_factor = _meta_crop_param->get_area_factor();
+    aspect_ratio_factor = _meta_crop_param->get_aspect_ratio();
+    x_drift_factor = _meta_crop_param->get_x_drift_factor();
+    y_drift_factor = _meta_crop_param->get_y_drift_factor();
+
+    //These Two Variable are suffincient for now
+    in_width = _meta_crop_param->in_width;
+    in_height = _meta_crop_param->in_height;
+    bool invalid_bboxes = true;
+    BoundingBoxCord crop_box, jth_box;
+    int bb_count;
+    for (int i = 0; i < _batch_size; i++)
+    {
+        bb_count = input_meta_data->get_bb_labels_batch()[i].size();
+        std::vector<int> labels_buf(bb_count);
+        memcpy(labels_buf.data(), input_meta_data->get_bb_labels_batch()[i].data(), sizeof(int) * bb_count);
+        std::vector<float> coords_buf(bb_count * 4);
+        memcpy(coords_buf.data(), input_meta_data->get_bb_cords_batch()[i].data(), input_meta_data->get_bb_cords_batch()[i].size() * sizeof(BoundingBoxCord));
+        crop_box.h = _crop_height_val[i];
+        crop_box.w = _crop_width_val[i];
+        crop_box.x = _x1_val[i];
+        crop_box.y = _y1_val[i];
+        int count = 1000;
+        int min_left = INT32_MAX, min_top = INT32_MAX, max_right = INT32_MIN, max_bottom = INT32_MIN;
+        while (--count)
+        {
+            for (int j = 0; j < _num_of_attempts; j++)
+            {
+                x_drift_factor->renew();
+                float factor = 0.5f;
+                auto w_factor = factor + (x_drift_factor->get() * (1 - factor));
+                crop_box.w = w_factor * in_width[i];
```

Comment:
Check the logic here.  Looks like crop_box.w is always more than 0.5*W. It needs to get a uniform random number from 0.1 to 1.